WireFoundation/Tests/WireFoundationTests/Crypto/AES256CryptoTests.swift

```diff
@@ -0,0 +1,251 @@
+            ciphertext: changedCiphertext,
+            key: key
+        )
+
+        // Then
+        XCTAssertNotEqual(plaintext, originalData)
+    }
+
+    // MARK: - Prefixed IV
+
+    func testEncryptDecryptWithPrefixedIV() throws {
+        // Given
+        let originalData = Scaffolding.originalData(from: "Hello, world")
+        let key = try Scaffolding.randomKey()
+
+        // When
+        let ciphertext = try AES256Crypto.encryptAllAtOnceWithPrefixedIV(
+            plaintext: originalData,
+            key: key
+        )
+
+        // Then
+        XCTAssertNotEqual(ciphertext, originalData)
+
+        // When
+        let plaintext = try AES256Crypto.decryptAllAtOnceWithPrefixedIV(
+            ciphertext: ciphertext,
+            key: key
+        )
+
+        // Then
+        XCTAssertEqual(plaintext, originalData)
+    }
+
+    func testEncryptDecryptWithPrefixedIV_IVIsRandomlyGenerated() throws {
+        // Given
+        let originalData = Scaffolding.originalData(from: "Hello, world")
+        let key = try Scaffolding.randomKey()
+
+        // When
+        let ciphertext1 = try AES256Crypto.encryptAllAtOnceWithPrefixedIV(
+            plaintext: originalData,
+            key: key
+        )
+
+        let ciphertext2 = try AES256Crypto.encryptAllAtOnceWithPrefixedIV(
+            plaintext: originalData,
+            key: key
+        )
+
+        // Then
+        XCTAssertNotEqual(ciphertext1, ciphertext2)
+    }
+
+    func testEncryptWithPrefixIV_DecryptWithoutPrefixedIV() throws {
+        // Given
+        let originalData = Scaffolding.originalData(from: "Hello, world")
+        let key = try Scaffolding.randomKey()
+
+        let ciphertext = try AES256Crypto.encryptAllAtOnceWithPrefixedIV(
+            plaintext: originalData,
+            key: key
+        )
+
+        // When
+        let invalidPlaintext = try AES256Crypto.decryptAllAtOnce(
+            ciphertext: ciphertext,
+            key: key
+        )
+
+        let validPlaintext = try AES256Crypto.decryptAllAtOnceWithPrefixedIV(
+            ciphertext: ciphertext,
+            key: key
+        )
+
+        // Then
+        XCTAssertNotEqual(invalidPlaintext, originalData)
+        XCTAssertEqual(validPlaintext, originalData)
+    }
+
+    // MARK: - Keys
+
+    func testKeyGeneration() throws {
+        // When
+        let key = try AES256Crypto.generateRandomEncryptionKey()
+
+        // Then
+        XCTAssertEqual(key.count, 32)
+    }
+
+    func testEncryption_InvalidKey() throws {
+        // Given
+        let originalData = Scaffolding.originalData(from: "Hello, world")
+        let key = try Scaffolding.randomInvalidKey()
+
+        do {
+            // When
+            _ = try AES256Crypto.encryptAllAtOnce(
+                plaintext: originalData,
+                key: key
+            )
+
+            XCTFail("expected an error")
+
+        } catch AES256CryptoError.invalidKeyLength {
```

Comment:
Maybe we can use `XCTAssertThrowsError` helper method from `WireTesting` in WireFoundation here

Reply:
done 6bf0dd395dac49f8b664ed435bacec1f65e2e22c